x-pack/plugins/ml/public/embeddables/anomaly_charts/anomaly_charts_react_container.tsx

```diff
@@ -0,0 +1,385 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { useCallback, useState, useMemo, useEffect, useRef } from 'react';
+import { EuiCallOut, EuiLoadingChart, EuiResizeObserver, EuiText } from '@elastic/eui';
+import type { Observable } from 'rxjs';
+import { combineLatest, tap, debounceTime, switchMap, skipWhile, startWith, of } from 'rxjs';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { throttle } from 'lodash';
+import { UI_SETTINGS } from '@kbn/data-plugin/common';
+import {
+  type MlEntityField,
+  type MlEntityFieldOperation,
+  ML_ANOMALY_THRESHOLD,
+} from '@kbn/ml-anomaly-utils';
+import { TimeBuckets } from '@kbn/ml-time-buckets';
+import { Subject, catchError } from 'rxjs';
+import useObservable from 'react-use/lib/useObservable';
+import type { TimeRange } from '@kbn/es-query';
+import type { InfluencersFilterQuery } from '@kbn/ml-anomaly-utils';
+import type { CoreStart } from '@kbn/core/public';
+import type {
+  AnomalyChartsEmbeddableCustomInput,
+  AnomalyChartsEmbeddableServices,
+  AnomalyChartsServices,
+  AnomalyChartsApi,
+} from '..';
+
+import { ExplorerAnomaliesContainer } from '../../application/explorer/explorer_charts/explorer_anomalies_container';
+import { ML_APP_LOCATOR } from '../../../common/constants/locator';
+import { optionValueToThreshold } from '../../application/components/controls/select_severity/select_severity';
+import { EXPLORER_ENTITY_FIELD_SELECTION_TRIGGER } from '../../ui_actions/triggers';
+import type { MlLocatorParams } from '../../../common/types/locator';
+import { getJobsObservable } from '../common/get_jobs_observable';
+import { OVERALL_LABEL, SWIMLANE_TYPE } from '../../application/explorer/explorer_constants';
+import { processFilters } from '../common/process_filters';
+import type { AppStateSelectedCells } from '../../application/explorer/explorer_utils';
+import {
+  getSelectionInfluencers,
+  getSelectionJobIds,
+  getSelectionTimeRange,
+} from '../../application/explorer/explorer_utils';
+import type { ExplorerChartsData } from '../../application/explorer/explorer_charts/explorer_charts_container_service';
+import type { MlStartDependencies } from '../../plugin';
+
+const RESIZE_THROTTLE_TIME_MS = 500;
+const FETCH_RESULTS_DEBOUNCE_MS = 500;
+
+export interface EmbeddableAnomalyChartsContainerProps
+  extends Partial<AnomalyChartsEmbeddableCustomInput> {
+  lastReloadRequestTime?: number;
+  api: AnomalyChartsApi;
+  id: string;
+  services: AnomalyChartsEmbeddableServices;
+  timeRange$: Observable<TimeRange | undefined>;
+  onRenderComplete: () => void;
+  onLoading: (v: boolean) => void;
+  onError: (error: Error) => void;
+}
+function useAnomalyChartsData(
```

Comment:
Nit: this might be worth breaking out into a separate file.

Reply:
Updated here https://github.com/elastic/kibana/pull/183456/commits/03b2ee89bb7165607e7cf3177c9f61af22c8468e